pkg/registry/registry.go

```diff
@@ -0,0 +1,118 @@
+package registry
+
+import (
+	"errors"
+	"os"
+	"time"
+)
+
+// RegistryType is an enum-like type that represents the type of registry
+// useful to conditionally observe the registry according to its type
+type RegistryType int
+
+const (
+	ConstantRegistryType RegistryType = iota
+	RemoteRegistryType
+)
+
+// Registry is the interface for the JSON registry
+// It is used to get the JSON bytes representing a set of endpoints to redirects
+type Registry interface {
+	RegistryType() RegistryType
+	GetJSON() ([]byte, error)
+}
+
+// Observe the registry for changes
+// returns json bytes and errors via channels
+func Observe(registry Registry, period time.Duration) (<-chan []byte, <-chan error, func()) {
+	jsonChan := make(chan []byte)
+	errorsChan := make(chan error)
```

Comment:
better to have only one channel of a struct containing the json blob and the error like:
```
make(chan, struct{
   json []byte
   err error
})
```

Give this type a good name :).